src/main/java/teammates/logic/core/EmailGenerator.java

```diff
@@ -171,6 +171,35 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructorReminders(
     }
     
     /**
+     * Generates the feedback session closed emails for the given {@code session}.
+     * @throws EntityDoesNotExistException
+     */
+    public List<EmailWrapper> generateFeedbackSessionClosedEmails(FeedbackSessionAttributes session) {
+
+        boolean isEmailNeededForStudents = false;
+        try {
+            isEmailNeededForStudents = fsLogic.isFeedbackSessionHasQuestionForStudents(
+                    session.getSessionName(), session.getCourseId());
+        } catch (EntityDoesNotExistException e) {
+            // TODO Auto-generated catch block
+        }
+        CourseAttributes course = coursesLogic.getCourse(session.getCourseId());
+        
+        List<InstructorAttributes> instructors = new ArrayList<InstructorAttributes>();
+        List<StudentAttributes> students = new ArrayList<StudentAttributes>();
+        if (!session.isPrivateSession()) {
+            instructors = instructorsLogic.getInstructorsForCourse(session
+                    .getCourseId());
+
+            students = isEmailNeededForStudents
+                    ? studentsLogic.getStudentsForCourse(session.getCourseId())
+                    : new ArrayList<StudentAttributes>();
```

Comment:
`students` is initialized with an empty list before. Rather than assigning the same thing, just do a conditional assign, i.e
```java
if (...) {
    students = ...;
}
```